pitest-entry/src/main/java/org/pitest/mutationtest/build/intercept/annotations/ExcludedAnnotationInterceptor.java

```diff
@@ -39,17 +40,56 @@ public void begin(ClassTree clazz) {
     this.skipClass = clazz.annotations().stream()
         .anyMatch(avoidedAnnotation());
     if (!this.skipClass) {
-      final List<Predicate<MutationDetails>> methods = clazz.methods().stream()
+      // 1. Collect methods with avoided annotations
+      final List<MethodTree> avoidedMethods = clazz.methods().stream()
           .filter(hasAvoidedAnnotation())
+          .collect(Collectors.toList());
+
+      final Set<String> avoidedMethodNames = avoidedMethods.stream()
+          .map(method -> method.rawNode().name)
+          .collect(Collectors.toSet());
+
+      // 2. Collect lambda methods with being inside avoided methods
+      final List<MethodTree> lambdaMethods = clazz.methods().stream()
+          .filter(MethodTree::isGeneratedLambdaMethod)
+          .filter(lambdaMethod -> {
+            String lambdaName = lambdaMethod.rawNode().name; // e.g., lambda$fooWithLambdas$0
+            String enclosingMethodName = extractEnclosingMethodName(lambdaName);
```

Comment:
The logic here is based solely on method name. If you were to add the other suggested tests they would fail as all lambdas in all the overloads of a method would be filtered if any one of the overloads was annotated.

Reply:
I have updated the tests. Also, the logic should be adjusted to overridden methods. Let me know if there's something I miss...Thanks

Reply:
Hmm interesting that the build is failing after 1m but locally my build is successful without any problems. 
```java
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for pitest-parent dev-SNAPSHOT:
[INFO] 
[INFO] pitest-parent ...................................... SUCCESS [  0.365 s]
[INFO] Build Config ....................................... SUCCESS [  0.295 s]
[INFO] pitest ............................................. SUCCESS [  6.108 s]
[INFO] pitest-entry ....................................... SUCCESS [ 51.068 s]
[INFO] pitest-html-report ................................. SUCCESS [  0.782 s]
[INFO] pitest-aggregator .................................. SUCCESS [  0.795 s]
[INFO] pitest-maven ....................................... SUCCESS [  5.872 s]
[INFO] pitest-command-line ................................ SUCCESS [  0.637 s]
[INFO] pitest-ant ......................................... SUCCESS [  0.626 s]
[INFO] pitest-java8-verification .......................... SUCCESS [  3.993 s]
[INFO] pitest-maven-verification .......................... SUCCESS [02:09 min]
[INFO] pitest-modern-verification ......................... SUCCESS [  0.670 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:20 min
[INFO] Finished at: 2024-11-08T10:39:43+01:00
[INFO] ------------------------------------------------------------------------
```

Reply:
I don't think overrides need to be considered, just overloads within the same class. Although it is possible to navigate the class hierarchy from the bytecode, it becomes quite expensive when there are a lot of mutants.

Unhelpfully, I used the word override when I meant overload earlier when describing the test cases.

The failing test is likely because the suite runs a matrix of different java versions. Sometimes the bytecode generated differs substantially with version. Tests need to be careful what they assert on in case they are making implicit assumptions.

Reply:
Updated for overloads. Tested also with my code and it seems to work fine. Build locally is passing now I have to figure it out to fix it also upstream :D